### ISSUE TITLE
Add migration with version number for custom add-on

### ISSUE THREAD

**Issue by 3D-I** [rmcgirr83/activity24hours#11]
Hi Rich, I need a migration which states "version 1.0.6" in this extension, for the add-on that I am creating. Since it seems to be quite easy, I am bound to code the add-on during tonight and forget about that once and for all. I will make use of ext.php in my add-on to check if your extension is enabled and the version number is correct, The add-on installs only if the criteria are met.

If you don't mind I am bound to create the PR on purpose,
I would like to know if you agree or have a better idea.

Thanks. 👍

**Comment by rmcgirr83** [comment]
If you are doing this 

> I will make use of ext.php in my add-on to check if your extension is enabled and the version number is correct, The add-on installs only if the criteria are met

 why do you need a migration file?

**Comment by 3D-I** [comment]
You mean I could read the metadata via extension manager uh?

**Comment by rmcgirr83** [comment]
Yep though I have never tried that approach before.

**Comment by 3D-I** [comment]
Yeah, same here. 
The idea is good, the extension manager reads the metadata right before the ext.php runs AFAIK.

Ok, I will do... in case I will ask in the writers' forum, thanks. 👍

**Comment by 3D-I** [comment]
To add, Javier taught me that way isn't 100% sure though. 

Checking if the ext is enabled and checking for a config version is more reliable at the moment, I faced this discussion in the validation queue for TPOTM, and I applied such approach for the DAE compatibility part of the code (another ext. of mine, validated)

**Comment by rmcgirr83** [comment]
Not all extensions have versions written to the database though... Like
this one

On Dec 27, 2017 3:54 PM, "3Di" <notifications@github.com> wrote:

To add, Javier taught me that way isn't 100% sure though.

Checking if the ext is enabled and checking for a config version is more
reliable at the moment, I faced this discussion in the validation queue for
TPOTM, and I applied such approach for the DAE compatibility part of the
code (another ext. of mine, validated)

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/rmcgirr83/phpBB-3.1-activity24hours/issues/11#issuecomment-354176310>,
or mute the thread
